lib/linked_list.rb

```diff
@@ -2,44 +2,106 @@
 
 class LinkedList
   attr_reader :head
-
+  
   def initialize
     @head = nil
   end
-
-  # Time complexity - ?
-  # Space complexity - ?
+  
+  # Time complexity - O(1)
+  # Space complexity - O(1)
   def add_first(data)
-
+    temp = Node.new(data)
+    temp.next = @head
+    @head = temp
   end
-
-  # Time complexity - ?
-  # Space complexity - ?
+  
+  # Time complexity - O(1)
+  # Space complexity - O(1)
   def get_first
-
+    if @head == nil
+      return nil
+    end
+    
+    return @head.data
   end
-
-  # Time complexity - ?
-  # Space complexity - ?
+  
+  # Time complexity - O(n)
+  # Space complexity - O(1)
   def length
-    return 0
+    if @head == nil
+      return 0
+    end
+    
+    length = 1
+    current = @head
+    
+    while current.next != nil
+      length += 1
+      current = current.next
+    end
+    return length 
   end
-
-  # Time complexity - ?
-  # Space complexity - ?
+  
+  # Time complexity - O(n)
+  # Space complexity - O(1)
   def add_last(data)
-
+    
+    temp = Node.new(data)
+    current = @head
+    
+    if @head == nil
+      @head = temp
+      return
+    elsif current.next == nil
+      current.next = temp
+      return
+    else
+      current = self.last_node
+      current.next = temp
```

Comment:
This seems a little overly complicated.  Can you think of a way to simplify this a bit?